### PR TITLE
Improve platform page card styles

### LINKED ISSUE #1685: Improve Platform Page Card Styles
# Task

The Platform Tools page at https://ocaml.org/docs/platform needs to be improved:

|Current|Planned Design|
|-|-|
|Screenshot 2023-10-19 at 13-58-09 OCaml Platform|Screenshot from 2023-10-19 13-58-57|

Here is [the Figma document for reference regarding the styles](https://www.figma.com/file/J0w4b9exaHhVpvabKoGqTn/Design-Github-Issues?type=design&node-id=326%3A20&mode=dev&t=2r15NvzWGSZIepb5-1).

How to solve this issue:

1. Assign Tailwind CSS styles in `platform.eml` to match the Figma design. Try to match the design as closely as you can with default Tailwind spacings.

### PR DESCRIPTION
**Pull request by FatumaA**
This closes #1685

This adjusts the platform page card sections styling to match the provided figma document.
It:
- abstracts some of the repeated template into a function within the same file. 
- adds border and hover border styles to the card
- adds padding and spacing within the card
- replaces the exisiting icon with the correct one
- adds correct color to the new icon
- adds new icon to the card and colors it to match rest of the new card styles

Here is a screenshot for the changes made:
 
Screenshot 2023-11-11 at 13 20 43

The hover effect:

Screenshot 2023-11-11 at 13 21 07

### CODE REVIEW

**Comment by FatumaA** [comment]
Hi @sabine,
This PR is almost done. 

I need some clarification on the icons. The tool icon in the figma document is not defined in the icons.eml file. Should I add it? Are there any guidelines on that? Or is it somewhere else and I just missed it?

**Comment by FatumaA** [comment]
> Hi @sabine, This PR is almost done.
> 
> I need some clarification on the icons. The tool icon in the figma document is not defined in the icons.eml file. Should I add it? Are there any guidelines on that? Or is it somewhere else and I just missed it?

I went with [Bootstrap tool icon](https://icons.getbootstrap.com/icons/tools/).

File: src/ocamlorg_frontend/pages/platform.eml [hunk 1]
```diff
@@ -7,6 +7,16 @@ let render_sidebar
     <%s! Learn_sidebar.render ~tutorials ~current_tutorial:active ~section:Learn_layout.Platform %>
   </div>
 
+let render_package_cards ~(item: Data.Tool.t) =
+  <a href="<%s item.source %>" class="flex gap-4 border-2 rounded-md p-4 h-28 hover:border-primary-600 hover:ring-4 hover:ring-primary-200">
+    <%s! Icons.tools "h-5 w-5 text-primary-600 self-start" %>
```
Comment (sabine) [inline src/ocamlorg_frontend/pages/platform.eml]: There's something off about the alignment of the icon with respect to the name of the item. In the design, it looks like the orange text and the icon are pretty much center aligned wrt each other. :thinking:

I think the simplest way to achieve this is probably to make the icon and name a flex-row, and then, have another flex-row below (with an empty div on the left that's the same width as the icon).

**Review by sabine — COMMENTED** [review]
We're using https://heroicons.com. I'll add a note to the `icons.eml` file, so it's more clear for the next person. Thanks for bringing that up! :+1:

File: src/ocamlorg_frontend/pages/platform.eml (hunk 1 above)
Comment (FatumaA) [reply src/ocamlorg_frontend/pages/platform.eml]: Done. Should be better now

**Comment by FatumaA** [comment]
> We're using https://heroicons.com. I'll add a note to the `icons.eml` file, so it's more clear for the next person. Thanks for bringing that up! 👍

replaced it 😌

**Comment by FatumaA** [comment]
This is ready @sabine

**Review by sabine — APPROVED** [review]
LGTM!

Thanks for the contribution @FatumaA! :+1: :heart:

**Pull request merged**